### ISSUE TITLE
inputs.procstat: measurement procstat_lookup missing the query information in its tags

### ISSUE THREAD

**Issue by sajoupa** [influxdata/telegraf#9793]
### Relevant telegraf.conf:

```toml
[[inputs.procstat]]
# Just a random example
  exe = "cron"

```

### System info:

```
Telegraf 1.20.0 (git: HEAD f489cff3)

Description:	Ubuntu 20.04.3 LTS
Release:	20.04
```

### Steps to reproduce:

1. Enable the procstat input plugin in telegraf 1.20.0, and specify the relevant selectors 
2. Observe the procstat_lookup measurement

### Expected behavior:

The query information should be in the measurement tags as explained in https://github.com/influxdata/telegraf/tree/master/plugins/inputs/procstat#metrics.

### Actual behavior:

The tags are missing.

### Additional info:
```
root@focal:~# grep -A 1 procstat /etc/telegraf/telegraf.conf
[[inputs.procstat]]
 exe = "cron"
```

Telegraf 1.19.3 (there's an `exe=cron` tag):
```
root@focal:~# /usr/bin/telegraf --config /etc/telegraf/telegraf.conf --input-filter procstat --test | grep '^> procstat_lookup'
2021-09-22T13:15:17Z I! Starting Telegraf 1.19.3
> procstat_lookup,exe=cron,host=focal,pid_finder=pgrep,result=success pid_count=1i,result_code=0i,running=1i 1632316518000000000
```

Telegraf 1.20.0 (no more `exe=cron` tag):
```
root@focal:~# /usr/bin/telegraf --config /etc/telegraf/telegraf.conf --input-filter procstat --test | grep '^> procstat_lookup'
2021-09-22T13:14:37Z I! Starting Telegraf 1.20.0
> procstat_lookup,host=focal,pid_finder=pgrep,result=success pid_count=1i,result_code=0i,running=1i 1632316478000000000
```

**Comment by sajoupa** [comment]
> I downloaded the 1.20 tarball from the releases section and was unable to reproduce this. Is there another setting in your config file that is required?

No, I just took the .deb from the github release page, installed it in a focal lxd without any previous telegraf installation, and updated telegraf.conf to add this procstat definition.
But as you can see in the output you pasted, `exe=cron` is also missing from procstat_lookup:
> procstat_lookup,host=x1c9,pid_finder=pgrep,result=success pid_count=1i,running=1i,result_code=0i 1632321620000000000

**Comment by powersj** [comment]
ah right, thanks - I had missed it was only in procstat row and not procstat_lookup

**Comment by powersj** [comment]
My initial look shows the following fairly recent MP made changes in this area: https://github.com/influxdata/telegraf/pull/9488

**Comment by powersj** [comment]
Looks like the way the [procstat_lookup tags](https://github.com/influxdata/telegraf/pull/9488/files#diff-cc14c88f3221cbdf2142e6f2519dfbea11c8974a3ad070ee74775ab632a996f2R169) were constructed was changed. It is now only going to be the `pid_finder` and `result` and not the higher-level general tags.

@reimda this seems like a regression that should be fixed. Thoughts?

**Comment by grimz-ly** [comment]
This is rather integral for how we've structured our service/process checks so guess I'll roll us all back for now until I see resolve on this thread.

**Comment by powersj** [comment]
Can one of you try the fix in #9808? There are artifacts that can be downloaded and tried. It would be great to try this with a larger config, as I only tried something very simple.

**Comment by sajoupa** [comment]
> Can one of you try the fix in #9808? There are artifacts that can be downloaded and tried. It would be great to try this with a larger config, as I only tried something very simple.

Thanks, I'll try and find some time tomorrow to test it. Our setup isn't extremely complex either (2 procstat inputs, with just one query pattern in each).
NB: we've also rolled back to using 1.19.3 in production for now (https://snapcraft.io/telegraf).

**Comment by grimz-ly** [comment]
the dev box I tested on only has two checks set on it, but seems that patched file did the trick. Something in my compile or ignorance where the version is missing at the end. thanks for the quick fix to this!

`# ./telegraf --config /etc/telegraf/telegraf.conf --config /etc/telegraf/telegraf.d/services.conf --test |grep procstat_lookup
2021-09-23T21:14:10Z I! Starting Telegraf 
> procstat_lookup,host=testlocal,pid_finder=pgrep,process_name=sshd,result=success,systemd_unit=sshd.service,user=root pid_count=1i,result_code=0i,running=1i 1632431651000000000
> procstat_lookup,host=testlocal,pid_finder=pgrep,process_name=httpd,result=success,systemd_unit=httpd.service,user=root pid_count=1i,result_code=0i,running=1i 1632431651000000000`

**Comment by sajoupa** [comment]
> Can one of you try the fix in #9808? There are artifacts that can be downloaded and tried. It would be great to try this with a larger config, as I only tried something very simple.

Looks good to me as well. On one of the affected hosts, with this new version the tags which were previously missing in 1.20.0 are now present.
Thanks !
